beacon_node/lighthouse_network/src/rpc/codec.rs

```diff
@@ -538,68 +544,150 @@ fn handle_rpc_request<E: EthSpec>(
-        ))),
-        SupportedProtocol::BlobsByRangeV1 => Ok(Some(InboundRequest::BlobsByRange(
-            BlobsByRangeRequest::from_ssz_bytes(decoded_buffer)?,
-        ))),
+                )?),
+            )))
+        }
+        SupportedProtocol::BlocksByRootV2 => {
+            metrics::inc_counter_vec_by(
+                &metrics::TOTAL_RPC_REQUESTS_BYTES_RECEIVED,
+                &["blocks_by_root"],
+                decoded_buffer.len() as u64,
+            );
+            Ok(Some(InboundRequest::BlocksByRoot(BlocksByRootRequest::V2(
+                BlocksByRootRequestV2 {
+                    block_roots: RuntimeVariableList::from_ssz_bytes(
+                        decoded_buffer,
+                        spec.max_request_blocks as usize,
+                    )?,
+                },
+            ))))
+        }
+        SupportedProtocol::BlocksByRootV1 => {
+            metrics::inc_counter_vec_by(
+                &metrics::TOTAL_RPC_REQUESTS_BYTES_RECEIVED,
+                &["blocks_by_root"],
+                decoded_buffer.len() as u64,
+            );
+            Ok(Some(InboundRequest::BlocksByRoot(BlocksByRootRequest::V1(
+                BlocksByRootRequestV1 {
+                    block_roots: RuntimeVariableList::from_ssz_bytes(
+                        decoded_buffer,
+                        spec.max_request_blocks as usize,
+                    )?,
+                },
+            ))))
+        }
+        SupportedProtocol::BlobsByRangeV1 => {
+            metrics::inc_counter_vec_by(
+                &metrics::TOTAL_RPC_REQUESTS_BYTES_RECEIVED,
+                &["blobs_by_range"],
+                decoded_buffer.len() as u64,
+            );
+            Ok(Some(InboundRequest::BlobsByRange(
+                BlobsByRangeRequest::from_ssz_bytes(decoded_buffer)?,
+            )))
+        }
         SupportedProtocol::BlobsByRootV1 => {
+            metrics::inc_counter_vec_by(
+                &metrics::TOTAL_RPC_REQUESTS_BYTES_RECEIVED,
+                &["blobs_by_root"],
+                decoded_buffer.len() as u64,
+            );
             Ok(Some(InboundRequest::BlobsByRoot(BlobsByRootRequest {
                 blob_ids: RuntimeVariableList::from_ssz_bytes(
                     decoded_buffer,
                     spec.max_request_blob_sidecars as usize,
                 )?,
             })))
         }
-        SupportedProtocol::DataColumnsByRangeV1 => Ok(Some(InboundRequest::DataColumnsByRange(
-            DataColumnsByRangeRequest::from_ssz_bytes(decoded_buffer)?,
-        ))),
-        SupportedProtocol::DataColumnsByRootV1 => Ok(Some(InboundRequest::DataColumnsByRoot(
-            DataColumnsByRootRequest {
-                data_column_ids: RuntimeVariableList::from_ssz_bytes(
-                    decoded_buffer,
-                    spec.max_request_data_column_sidecars as usize,
-                )?,
-            },
-        ))),
+        SupportedProtocol::DataColumnsByRangeV1 => {
```

Comment:
There's a lot of repetition in this match, you can turn `SupportedProtocol` into a str and have a single register metric call